include/ultra64/abi.h

```diff
@@ -330,14 +337,21 @@ typedef short ENVMIX_STATE[40];
         _a->words.w1 = _SHIFTL(dmemi, 16, 16) | _SHIFTL(dmemo, 0, 16);     \
 }
 
-#define aLoadBuffer(pkt, s, d, c)                                      \
-{                                                                      \
-        Acmd *_a = (Acmd *)pkt;                                        \
-                                                                       \
-        _a->words.w0 = (_SHIFTL(A_LOADBUFF, 24, 8) |                   \
-                        _SHIFTL((c) >> 4, 16, 8) | _SHIFTL(d, 0, 16)); \
-        _a->words.w1 = (u32)(s);                                       \
-}
+/*
+ * Loads a buffer from RDRAM to DMEM.
+ *
+ * @param pkt pointer to a Acmd command
+ * @param ramAddrSrc RDRAM source address
```

Comment:
Maybe worth mentioning this can be any of a physical address, KSEG0 or KSEG1, the highest byte is removed in the microcode. The same can be said for `aLoadBuffer`

Reply:
What does the microcode use? (I mean, K0, K1?)
It may be easier to phrase it like "only the three lower bytes are used as an offset into KSEG0" if something like that happens idk

Reply:
The only thing microcodes need RDRAM addresses for are DMA, which requires physical addresses. Microcodes usually have the conversion to physical addresses built into them so that these details are hidden to the code preparing task data (i.e. Acmd buffers) on the CPU

Reply:
Remind me of the terminology again, does `RDRAM` only include `KSEG0` or both `KSEG0/1`? Or are we really just generalizing it to any physical address? Should I also comment directly that the highest byte is removed, as to why any physical address works?

Reply:
RDRAM is physical addresses `0x00000000` to `0x03EFFFFF`, in practice you only care about `0x00000000` to `0x00800000` as everything else reads 0 and ignores writes.
KSEG0 is a mirror of physical addresses starting at `0x80000000`. Accesses go through the CPU cache.
KSEG1 is a mirror of physical addresses starting at `0xA0000000`. Accesses are direct, no caching is done.

The CPU can only access physical addresses through their mirrors in KSEG0 and KSEG1. Not all of KSEG0 and KSEG1 is mapped to RDRAM, only the addresses inside the RDRAM address range above maps there.
DMA controllers like the RSP DMA controller expect physical addresses when interacting with RDRAM.

I'm not sure if it's worth mentioning the exact implementation, that the highest byte is removed. In general the highest byte will always just be the bits determining whether it's KSEG0/KSEG1/physical, everything else will be 0 as the amount of RDRAM available doesn't extend that high. The reason why I think it's worth mentioning which addresses work is because this doesn't work for any other kind of address, like TLB-mapped addresses which aren't 1:1 with physical addresses (modulo the top few bits)